stretch_core/stretch_core/stretch_driver.py

```diff
@@ -617,32 +622,22 @@ def ros_setup(self):
                                                    '/runstop',
                                                    self.runstop_service_callback)
 
+        timer_period = 1.0 / self.joint_state_rate
+        self.timer = self.create_timer(timer_period, self.command_mobile_base_velocity_and_publish_state)
+
     def parameter_callback(self, parameters):
         self.get_logger().warn('Dynamic parameters not available yet')
 
-    ########### MAIN ############
-    def main(self):
-        self.ros_setup()
-
-        try:
-            # start loop to command the mobile base velocity, publish
-            # odometry, and publish joint states
-            while rclpy.ok():
-                self.command_mobile_base_velocity_and_publish_state()
-                self.command_base_velocity_and_publish_joint_state_rate.sleep()
-        except (KeyboardInterrupt, ThreadServiceExit):
-            self.robot.stop()
-
 
 def main():
-    rclpy.init()
-    node = StretchBodyNode()
-    thread = threading.Thread(target=rclpy.spin, args=(node, ), daemon=True)
-    thread.start()
-
-    node.main()
-    rclpy.shutdown()
-    thread.join()
+    try:
+        rclpy.init()
+        executor = MultiThreadedExecutor(num_threads=2)
```

Comment:
Is there any particular reason we use 2 threads?

Reply:
For now, its just N > 1.